code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -901,6 +914,11 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
 		overlays_standing[SUIT_LAYER] = standing
 
 	apply_overlay(SUIT_LAYER)
+	
+	update_inv_gloves()
```

Comment:
Also for a future PR. Change the way this is done. Because now it is possible you update the gloves/shoes/etc multiple times in one call.
Something like how #17943 handles it would work